src/main/java/seedu/duke/modules/ModuleList.java

```diff
@@ -106,4 +108,36 @@ public Map<Integer, ArrayList<Module>> groupModulesBySemester() {
 
         return moduleBySemMap;
     }
+
+    public boolean containsModule(String moduleCode) {
+        for (Module takenModule : takenModuleList) {
+            if (moduleCode.equals(takenModule.getModuleCode())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public Integer getUEModuleMCToComplete() {
+        int totalUnrestrictedElectivesMCs = 40;
+        int takenUnrestrictedElectiveMCs = 0;
+        for (Module takenModule : takenModuleList) {
+            try {
+                CEGModules.mapStringToEnum(takenModule.getModuleCode());
+            } catch (IllegalArgumentException e) {
+                takenUnrestrictedElectiveMCs += takenModule.getModuleMC();
+            }
+        }
+        return Math.max(totalUnrestrictedElectivesMCs - takenUnrestrictedElectiveMCs, 0);
+    }
```

Comment:
Good work in trying to display unrestricted electives MCs!

One issue with this, currently your UE MC remaining calculation relies on matching the courseCode directly with what is in the list which is mostly sufficient.

However specifically for a normal CEG student, the GE pillar courses are quite varied. 

For your method, adding something like "GEC1015" or "USE2310" **will both count under UE**, even though taking **either one of it** should count under "Cultures and Connections (GEC)" pillar. 

If we take both "GEC1015" and "USE2310", one of them will be considered under UE, while the other will be considered under GEC pillar.

Can read the whole GE pillar list from here: https://www.nus.edu.sg/registrar/academic-information-policies/undergraduate-students/general-education/list-of-courses-approved-under-the-ge-pillars

If this is too troublesome, then i think its better to just not calculate the UE's remaining, and just display the graduation list as it is

Reply:
Also, for CEG students. **CDE2501** specifically will not count under Singapore Studies (GESS) pillar. It will count as part of Common Curriculum Requirements.